src/shm_sub.c

```diff
@@ -2270,16 +2285,19 @@ sr_shmsub_rpc_internal_call_callback(sr_conn_ctx_t *conn, const struct lyd_node
         return err_info;
     }
 
-    /* collect all required modules */
-    LY_LIST_FOR(lyd_child(input), child) {
+    /* get modules which should be resetted */
+    if ((err_info = sr_lyd_find_xpath(input, "modules/module", &set))) {
```

Comment:
The sets are not really required but I can accept it.

Reply:
not sure whether I will find the time this week to rework that, so if you are fine with the current approach, we can leave it as it is

Reply:
I am in no hurry and `lyd_find_xpath()` really seems unnecessary in this case.

Reply:
Updated :)

I guess using the set in `sysrepo.c` is fine? There its quite convenient to check for the absence of nodes.